addons/product/pricelist.py

```diff
@@ -483,19 +352,21 @@ def _check_margin(self, cr, uid, ids, context=None):
             digits_compute= dp.get_precision('Product Price'), help='Specify the minimum amount of margin over the base price.'),
         'price_max_margin': fields.float('Max. Price Margin',
             digits_compute= dp.get_precision('Product Price'), help='Specify the maximum amount of margin over the base price.'),
-        'company_id': fields.related('price_version_id','company_id',type='many2one',
-            readonly=True, relation='res.company', string='Company', store=True)
+        'company_id': fields.related('pricelist_id','company_id',type='many2one',
+            readonly=True, relation='res.company', string='Company', store=True),
+        'date_start': fields.date('Start Date', help="Starting date for the pricelist item validation"),
+        'date_end': fields.date('End Date', help="Ending valid for the pricelist item validation"),
+        'compute_price': fields.selection([('fixed', 'Fix Price'), ('percentage', 'Percentage (discount)'), ('formula', 'Formula')], select=True, default='fixed'),
+        'fixed_price': fields.float('Fixed Price'),
+        'percent_price': fields.float('Percentage Price'),
+        'price_name': fields.function(_price_list_item_name, type="char", string='Price', store=True, multi='price_item', help="Explicit rule name for this pricelist line."),
+        'currency_id': fields.many2one('res.currency', 'Currency', required=True),
```

Comment:
currency_id should NOT be in the pricelist.item, but on the pricelist itself. (all lines of a pricelist are in the same currency) The currency is defined by the pricelist. So, this field should be removed.